src/common/components/ChainSelector.tsx

```diff
@@ -0,0 +1,191 @@
+"use client"
+
+import React, { useEffect } from "react";
+import { ethers } from 'ethers';
+import { switchChain } from '@wagmi/core'
+import {
+    getBalance,
+} from '@wagmi/core'
+
+import {
+    CheckIcon,
+    CaretSortIcon,
+} from "@radix-ui/react-icons"
+
+import { cn } from "@/lib/utils"
+import { Button } from "@/components/ui/button"
+
+import { config } from "@/common/helpers/rainbowkit";
+
+import {
+    Command,
+    CommandGroup,
+    CommandItem,
+} from "@/components/ui/command"
+import {
+    Popover,
+    PopoverContent,
+    PopoverTrigger,
+} from "@/components/ui/popover"
+import { arbitrum, base, optimism, zora } from "viem/chains"
+import { useAccount } from "wagmi"
+
+interface Chain {
+    value: string;
+    label: string;
+    icon: string;
+    chainId: number;
+    balance: undefined | bigint;
+}
+
+let defaultChains: Chain[] = [
```

Comment:
instead of hard-coding the chains here, we want the dropdown to show cross-chain tokens. 
you can get a list of relevant tokens for each users from this API:
https://docs.paywithglide.xyz/api-reference/list-payment-options

this PR https://github.com/hellno/herocast/pull/222 or branch `usewalletkit:main` has basic setup for glide already done. you can merge with this branch so you can just re-use the same client instance to fill the tokens into the dropdown to personalize the experience for each user.